lib/utils.pm

```diff
@@ -902,6 +903,13 @@ sub install_all_from_repo {
     assert_script_run($exec_str);
 }
 
+sub gather_pidstat_data {
+    my ($pidstat_file) = @_;
+
+    script_run('pkill pidstat');
+    upload_logs($pidstat_file);
+}
+
```

Comment:
I am afraid this is getting way too complicated. That baseclass would be just for two tests as that `post_fail_hook` is specific to those two tests, not even for all toolchain tests. (It may end up as an excuse but the PID statistics data are hoped to be just a temporary thing.)